src/main/java/com/alvarium/serializers/AlvariumPersistence.java

```diff
@@ -0,0 +1,20 @@
+package com.alvarium.serializers;
+
+import com.alvarium.PublishWrapper;
+import com.alvarium.contracts.Annotation;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.time.ZonedDateTime;
+
+public class AlvariumPersistence {
+
+
+    public static final Gson GSON = new GsonBuilder()
```

Comment:
It doesnt that much, I could create a `getGson()` but a Gson object is not something we gonna to mutate, so why not place it in a static place and access it.

I made the GSON instance private, and now expose three static functions from the classs : 
```java
public class PeristenceFunctions {

  private static final Gson GSON = ...;

  public static String serializeWrapper(PublishWrapper wrapper) {
    return GSON.toJson(wrapper);
  }

  public static String serializeAnnotation(Annotation annotation) {
    return GSON.toJson(annotation);
  }

  public static Annotation deserializeAnnotation(String jsonAnnotation) {
    return GSON.fromJson(jsonAnnotation, Annotation.class);
  }

}
```

Reply:
Well the reason behind my suggestion was that we could just do something like this: 

```java

public class JSONSerializer<T> {

  private final Gson gson;
  
  public JSONSerializer() {
      gson = new GsonBuilder()
        .registerTypeAdapter(PublishWrapper.class, new PublishWrapperConverter())
        .registerTypeAdapter(ZonedDateTime.class, new ZonedDateTimeConverter())
        .registerTypeAdapter(Annotation.class, new AnnotationConverter())
        .disableHtmlEscaping()
        .create();
  }

  public String toJson(T serializable) {
    return this.gson.toJson(serializable);
  }


  public T fromJson(String json, java.lang.Class<T> classOfT) {  // maybe there is a way to skip the second parameter, I haven't dug deep enough to check
    return this.gson.fromJson(json, classOfT);
  }

}
```

and to use it we can just do `new JSONSerializer<Annotation>().toJson(annotation)` or `new JSONSerializer<Annotation>().fromJson(json, Annotation.class)`. No static instances of anything, and I assume no detriment to performance.

Thoughts ?

Reply:
That does make sense. I like the idea of enforcing the use of the serializer to only specific types in order to avoid performance issues, so I am sold on that part. But you do realize this will require a fair bit of coupling between the serializers package and any serializable class... Do you think it is possible to move the serialization logic as a member of the respective class ? i.e., something on the lines of `String json = publishWrapper.toJson()` while keeping all performance improvements ?

Reply:
Sorry i don't understand, do you want to move the functions of the `PersistanceFunctions.serializeX` to `X.toJson()` ? If yes then the methods already exists in `PublishWrapper` and `Annotation`. 

Or do you want to have a GSON instance in PublishWrapper and Annotation and then those gson are specialized to the they are associated ? If yes then would be like it is done currently, and this will force you to ensure that the changes you make are set in all Gsons (if you add an adpater then you'll need to not forget to place it in all Gson) this looks like code duplication i think 🤔

Reply:
Yeah sure no problem !